server/http/custom_routes.go

```diff
@@ -177,31 +124,36 @@ func (c *customController) GetTestVersionDefinitionFile(w http.ResponseWriter, r
 	w.Write(result.Body.([]byte))
 }
 
-func (c *customController) GetTransactions(w http.ResponseWriter, r *http.Request) {
-	query := r.URL.Query()
-	takeParam, err := parseInt32Parameter(query.Get("take"), false)
-	if err != nil {
-		c.errorHandler(w, r, &openapi.ParsingError{Err: err}, nil)
-		return
-	}
-	skipParam, err := parseInt32Parameter(query.Get("skip"), false)
-	if err != nil {
-		c.errorHandler(w, r, &openapi.ParsingError{Err: err}, nil)
-		return
-	}
-	queryParam := query.Get("query")
-	sortByParam := query.Get("sortBy")
-	sortDirectionParam := query.Get("sortDirection")
-	result, err := c.service.GetTransactions(r.Context(), takeParam, skipParam, queryParam, sortByParam, sortDirectionParam)
-	// If an error occurred, encode the error with the status code
-	if err != nil {
-		c.errorHandler(w, r, err, &result)
-		return
-	}
-	res := result.Body.(paginated[openapi.Transaction])
+func paginatedEndpoint[T any](
+	f func(c context.Context, take, skip int32, query string, sortBy string, sortDirection string) (openapi.ImplResponse, error),
+	errorHandler openapi.ErrorHandler,
+) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		query := r.URL.Query()
+		takeParam, err := parseInt32Parameter(query.Get("take"), false)
+		if err != nil {
+			errorHandler(w, r, &openapi.ParsingError{Err: err}, nil)
+			return
+		}
+		skipParam, err := parseInt32Parameter(query.Get("skip"), false)
+		if err != nil {
+			errorHandler(w, r, &openapi.ParsingError{Err: err}, nil)
+			return
+		}
+		queryParam := query.Get("query")
+		sortByParam := query.Get("sortBy")
+		sortDirectionParam := query.Get("sortDirection")
+		result, err := f(r.Context(), takeParam, skipParam, queryParam, sortByParam, sortDirectionParam)
+		// If an error occurred, encode the error with the status code
+		if err != nil {
+			errorHandler(w, r, err, &result)
+			return
+		}
+		res := result.Body.(paginated[T])
 
-	w.Header().Set("X-Total-Count", strconv.Itoa(res.count))
-	openapi.EncodeJSONResponse(res.items, &result.Code, w)
+		w.Header().Set("X-Total-Count", strconv.Itoa(res.count))
+		openapi.EncodeJSONResponse(res.items, &result.Code, w)
+	}
```

Comment:
Added this so we don't have to keep copy and pasting the same code for each endpoint